tests/test_gocamgen.py

```diff
@@ -59,3 +59,18 @@ def test_get_with_froms():
     ca_set.find_or_create_collapsed_association(go_assoc, with_from=header1)
     ca_set.find_or_create_collapsed_association(go_assoc, with_from=header2)
     assert len(ca_set.collapsed_associations) == 1 and ca_set.collapsed_associations[0].with_froms == ["FAKE:12345", "MGI:MGI:1915834"]
+
+
+def test_ref_picker():
+    test_refs = [
+        "GO_REF:0000483",
+        "doi:485930",
+        "WB_REF:WBPaper00003384",
+        "PMID:9834189",
+    ]
+    result = gocamgen.ReferencePreference.pick(test_refs)
+    assert result == "PMID:9834189"
+
+    test_refs = ["ZFIN:ZDB-PUB-170709-3"]
+    result = gocamgen.ReferencePreference.pick(test_refs)
+    assert result == "ZFIN:ZDB-PUB-170709-3"
```

Comment:
Question: what would happen if there were no refs? Is that a possibility?

Reply:
There would be an `IndexError` for empty list. But there should always be _something_ in references once it got to this point, otherwise it would fail elsewhere in gocamgen or get chucked out by the GPAD parser earlier.